File: output/gpt_4o/avatar/direct_translation/java/python/codeforces_299_A.py

import sys

def main():
    input = sys.stdin.read
    data = input().split()
    
    n = int(data[0])
    a = list(map(int, data[1:]))
    
    a.sort()
    min_value = a[0]
    
    for value in a:
        if value % min_value != 0:
            print(-1)
            return
    
    print(min_value)

if __name__ == "__main__":
    main()

File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/python/java/codeforces_299_A.java

import java.util.Arrays;
import java.util.Scanner;

public class codeforces_299_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        scanner.nextLine(); // Read and discard the first line of input
        String[] input = scanner.nextLine().split(" ");
        int[] a = Arrays.stream(input).mapToInt(Integer::parseInt).toArray();
        Arrays.sort(a);
        boolean anyNotDivisible = false;
        for (int x : a) {
            if (x % a[0] != 0) {
                anyNotDivisible = true;
                break;
            }
        }
        System.out.println(anyNotDivisible ? a[0] : -1);
    }
}

